src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -89,16 +102,35 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
                 fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.LocalPath);
                 return fileDocumentIds;
             }
+            else if (documentUri.Scheme == SourceGeneratedFileScheme)
+            {
+                // In this case, the "host" portion is just the ProjectId directly.
+                var projectId = ProjectId.CreateFromSerialized(Guid.ParseExact(documentUri.Host, SourceGeneratedGuidFormat));
+
+                // The AbsolutePath will consist of a leading / to ignore, then the GUID that is the DocumentId, and then another slash, then the hint path
+                var slashAfterId = documentUri.AbsolutePath.IndexOf('/', startIndex: 1);
```

Comment:
A little nervous about the slashes here (e.g. if they get escaped or if they get reversed).  I'm wondering if query parameters might make more sense to use

Reply:
I guess I'm not terribly worried, only because if slashes are getting corrupted, then we'd be equally broken with regular file:// URIs, no?

Reply:
I now switched over to query parameters, simply because I need to pack more things in, and also as I discovered the local path portion is shown in VS Code, so it's strange if we stick GUIDs in there.